packages/client/hooks/useTrebuchetEvents.ts

```diff
@@ -55,16 +56,24 @@ const useTrebuchetEvents = () => {
 
     const onReconnected = () => {
       setConnectedStatus(true)
-      atmosphere.retries.forEach((retry) => retry())
       atmosphere.eventEmitter.emit('removeSnackbar', ({key}) => key === 'offline')
     }
 
     const onData = async (payload: string | object) => {
       // hacky but that way we don't have to double parse huge json payloads. SSE graphql payloads are pre-parsed
       const obj = typeof payload === 'string' ? JSON.parse(payload) : payload
-      if (obj.version && obj.version !== __APP_VERSION__ && 'serviceWorker' in navigator) {
-        const registration = await navigator.serviceWorker.getRegistration()
-        registration?.update().catch()
+
+      // obj.version is sent once on connection, if we have disconnects logged, it's reconnect
+      const isReconnect = obj.version && recentDisconnectsRef.current.length > 0
```

Comment:
+1 nit-- since checking for `obj.version` is in both branches of the conditional, we can clean it up a little like

```ts
if (obj.version) {
  // this is the initial payload
  if (obj.version !== serverVersionRef.current) {
    // handle upgrade
  } else if (recentDisconnectsRef.current.length > 0) {
    // handle reconnect
  }
}